api/main.py

```diff
@@ -202,6 +217,96 @@ async def recommend_file(request: Request, pdf: UploadFile = File(...), size: in
     return JSONResponse(content=json_compatible_item_data)
 
 
+@app.post(
+    "/recommend_text",
+    dependencies=[Depends(RateLimiter(times=rate_times, seconds=rate_seconds))],
+)
+async def recommend_text(
+    request: Request,
+    sow: Sow,
+    size: int = 10,
+    index_types=["flat", "flat_sklearn"],
+    vectorizer_types=["tf_idf"],
+):
+    """Given an input of Statement of Work as text,
+    return a JSON of recommended standards.
+    """
+    start = time.time()
+    in_text = sow.text_field
+    # df_file = "data/feather_text"
+    list_of_texts = extraction.get_list_of_text(es)
+    vectorizers, vector_storage, vector_indexes = extraction.load_into_memory(
+        index_types, vectorizer_types
+    )
+    list_of_predictions, scores = extraction.predict(
+        in_text,
+        size,
+        vectorizers,
+        vector_storage,
+        vector_indexes,
+        list_of_texts,
+    )
+    output = {}
+    for i, prediction_id in enumerate(list_of_predictions):
+        res = es.search(
+            index=idx_main,
+            body={"size": 1, "query": {"match": {"doc_number": prediction_id}}},
+        )
+        for hit in res["hits"]["hits"]:
+            results = hit["_source"]
+        output[i] = results
+        output[i]["similarity"] = scores[i]
+    json_compatible_item_data = jsonable_encoder(output)
+    log_stats(request, data=in_text)
+    print(f"{time.time() - start}")
+    return JSONResponse(content=json_compatible_item_data)
+
+
+@app.post(
+    "/recommend_file",
+    dependencies=[Depends(RateLimiter(times=rate_times, seconds=rate_seconds))],
+)
+async def recommend_file(
+    request: Request,
+    pdf: UploadFile = File(...),
+    size: int = 10,
+    index_types=["flat", "flat_sklearn"],
+    vectorizer_types=["tf_idf"],
+):
+    """Given an input of a Statement of Work as a PDF,
+    return a JSON of recommended standards.
+    """
+    print("File received")
+    in_text = extract_prep.parse_text(pdf)
+    # df_file = "data/feather_text"
+    list_of_texts = extraction.get_list_of_text(es)
+    vectorizers, vector_storage, vector_indexes = extraction.load_into_memory(
+        index_types, vectorizer_types
+    )
+    list_of_predictions, scores = extraction.predict(
+        in_text,
+        size,
+        vectorizers,
+        vector_storage,
+        vector_indexes,
+        list_of_texts,
+    )
+    output = {}
+    for i, prediction_id in enumerate(list_of_predictions):
+        res = es.search(
+            index=idx_main,
+            body={"size": 1, "query": {"match": {"doc_number": prediction_id}}},
+        )
+        for hit in res["hits"]["hits"]:
+            results = hit["_source"]
+        output[i] = results
+        output[i]["similarity"] = scores[i]
+    json_compatible_item_data = jsonable_encoder(output)
+    log_stats(request, data=in_text)
+    print(f"{time.time() - start}")
+    return JSONResponse(content=json_compatible_item_data)
```

Comment:
@vlall this is repeated in recommend text. Let us create a separate function with this code and use it in both endpoints!